src/actors/cficaches.rs

```diff
@@ -199,7 +204,7 @@ impl CfiCacheActor {
     pub fn fetch(
         &self,
         request: FetchCfiCache,
-    ) -> impl Future<Item = Arc<CfiCacheFile>, Error = Arc<CfiCacheError>> {
+    ) -> Pin<Box<dyn Future<Output = Result<Arc<CfiCacheFile>, Arc<CfiCacheError>>>>> {
```

Comment:
The clippy warning here is a good one. Symbolicator often uses such a type:

```rust
type SharedResult<T, E> = Result<Arc<T>, Arc<E>>;
pub fn fetch(...) -> Box<Pin<SharedResult<CfiCacheFile, CfiCacheError>>>;
```

Alternatively, probably for our async traits, this one may be a better fit:

```rust
type AsyncResult<T, E> = Pin<Box<dyn Future<Output Result<T, E>>>>;
pub fn fetch(...) -> AsyncResult<Arc<CfiCacheFile>, Arc<CfiCacheError>>;
```

Reply:
I'm not a big fan of type aliases they often make things harder to read when coming back to the code.  In this case it's all decoration and the real signature is only the `Result` part.  But if clippy says I'll obey, I think the `AsyncResult` is my preferred option, or maybe even more general `PinedFuture<Result<...>>` (though given `FutureExt::boxed()` naming maybe `BoxedFuture` is a better name)

Reply:
I think being opinionated with these types and having a semantic `AsyncResult` can be a good thing. I've made good experience with that in Relay, since they are quick to learn and then improve readability while also helping to streamline the code.